hadoop-ozone/dist/src/main/smoketest/s3/maxkeys_validation.robot

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+*** Settings ***
+Documentation     S3 max-keys validation test for negative and zero values
+Library           OperatingSystem
+Library           String
+Resource          ../commonlib.robot
+Resource          commonawslib.robot
+Test Timeout      3 minutes
+Suite Setup       Setup s3 tests
+
+*** Variables ***
+${ENDPOINT_URL}   http://s3g:9878
+${BUCKET}         generated
+
+*** Test Cases ***
+
+List objects with negative max-keys should fail
+    ${result} =    Execute AWSS3APICli and checkrc    list-objects-v2 --bucket ${BUCKET} --max-keys -1    255
+    Should Contain    ${result}    InvalidArgument
+
+List objects with zero max-keys should fail
+    ${result} =    Execute AWSS3APICli and checkrc    list-objects-v2 --bucket ${BUCKET} --max-keys 0    255
```

Comment:
If you make the server-side max configurable, you can set it low for this test and test server-side enforcement as well.

Reply:
Sure, just add two robot tests for testing max-keys is lower or higher than maxKeysLimit scenarios.